src/nxt_listen_socket.c

```diff
@@ -112,15 +138,63 @@ nxt_listen_socket_create(nxt_task_t *task, nxt_mp_t *mp,
 
         nxt_sockaddr_text(sa);
 
-        (void) unlink(sa->u.sockaddr_un.sun_path);
```

Comment:
Well, I think the previous one is good enough.

If you think it's possible that the temp file can be existing like created manually, it happens very rarely.
We can just to use `unlink(sa->u.sockaddr_un.sun_path)`.

My thought is based on we need to do more in the future about this feature to clean up sock files.
So we can just keep the previous code here.

Your way may be used in the future. But now I prefer not to introduce it.

Reply:
This improvement was motivated by this comment:
https://github.com/nginx/unit/pull/649#issuecomment-1043331906

Yes, this function does not implement a check to see if the socket is abandoned or not. But for this, as recommended everywhere and for similar use in linux, a lock is required, which I feel we will not get to soon.

Reply:
In addition to my previous answer, I want to note that ignoring the error handling of `(void) unlink` here may cause `bind` to fail later and throw an error (`EADDRINUSE`) not related to the cause of the failure (e.g. `EACCES`), but related to the consequence of using `bind` after `(void) unlink`.

Reply:
Good morning @mar0x, 

Since you suggested this feature earlier, please let me know what you think of my proposed implementation. Thank you!

Reply:
After a little analysis of the code section you specified. It seems useless to me, because if the main task of `connect` in this code is to check if the `sockaddr` address is busy, then it is too late and not functional, which is actually confirmed by the current `EADDRINUSE` error from `bind`. What do you think about this?